docs/proposals/node-group-management.md

```diff
@@ -0,0 +1,262 @@
+1. Node that is not in any nodegroup which the pod should be scheduled to
+2. Node that is in the nodegroup which has already had enough replica number of the pod
+
+#### Score
+`GroupSchedulingExtender` will give a score for each candidate node that has passed the filter, and then send the scores back to the `kube-scheduler`. `kube-scheduler` will combine all scores(from built-in score plugins and the `GroupSchedulingExtender`), and finally pick one which has the highest score. The score method of `GroupSchedulingExtender` is mainly based on the difference between current pod replica number and desired pod replica number. In other words, a node will get a higher score if it's in a nodegroup with greater value of `DesiredPodReplicaNumber - CurrentPodReplicaNumber`.  
+> Note:  
+> The score method of `GroupSchedulingExtender` is at the nodegroup level, which means all nodes in the same nodegroup will get the same score from `GroupSchedulingExtender`. The prioritization among these nodes depends on other score plugins in `kube-scheduler`, such as `NodeAffinityPriority`.
+
+
+### GroupManagementControllerManager 
+`GroupManagementControllerManager` contains two controllers, called `NodeGroupController` and `PropagationPolicyController`.
+
+`NodeGroupController` is responsible for reconciling the NodeGroup Object which will collect nodes belonging to the NodeGroup according to node names and labels and fill the status field.
+
+`PropagationPolicyController` is responsible for reconciling the current replica number of pods in each nodegroup with the desired distribution status which is defined in the PropagationPolicy. It will evict pods in nodegroups where they should not run and also evict pods in nodegroups when they are redundant(which means the current replica number of pods in the nodegroup exceeds what the policy desires).   
+
+### NodeGroup API
+NodeGroup represents a group of nodes that has the same labels.
+```go
+// NodeGroup is the Schema for the nodegroups API
+type NodeGroup struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec represents the specification of the desired behavior of member nodegroup.
+	// +required
+	Spec NodeGroupSpec `json:"spec"`
+
+	// Status represents the status of member nodegroup.
+	// +optional
+	Status NodeGroupStatus `json:"status,omitempty"`
+}
+
+// NodeGroupSpec defines the desired state of NodeGroup
+type NodeGroupSpec struct {
+	// Nodes contains names of all the nodes in the nodegroup.
+	// +optional
+	Nodes []string `json:"nodes,omitempty"`
+
+	// MatchLabels match the nodes that have the labels
+	// +optional
+	MatchLabels map[string]string `json:"matchLabels,omitempty"`
+}
+
+// NodeGroupStatus defines the observed state of NodeGroup
+type NodeGroupStatus struct {
+	// ContainedNodes represents names of all nodes the nodegroup contains.
+	// +optional
+	ContainedNodes []string `json:"containedNodes,omitempty"`
+}
```

Comment:
The nodegroup defination is still needed, let's use it :) @lonelyCZ @Congrool @vincentgoat

Reply:
I'll update this proposal with the new design and put the `EdgeApplication` and `NodeGroup` API in it.